### ISSUE TITLE
release-22.1: descriptor not dropped for temp view after session ends

### ISSUE THREAD

**Issue by andyyang890** [cockroachdb/cockroach#108751]
This issue doesn't seem to be present in v22.2 or v23.1 (or master), but opening it anyway so we have a record of it and so we can determine when/how it was fixed. Also, since descriptors are persisted, there could theoretically be leftover descriptors when a user upgrades to newer versions without the issue.

**Steps to reproduce**

1. Start a session and run:
```
SET experimental_enable_temp_tables TO on;

CREATE TABLE t (x int);

CREATE TEMP VIEW t_view (x) AS SELECT x FROM t;

SELECT * FROM system.namespace; -- note the ID for the temp view
```

2. End the session and start a new session. Then run:
```
SELECT * FROM system.descriptor;
```

3. We expect the temp view descriptor to not be here, but it still is. This is problematic since the namespace entry for the temporary schema it was a part of has been dropped.

Jira issue: CRDB-30622

**Comment by andyyang890** [comment]
Oh this issue might be related: https://github.com/cockroachdb/cockroach/issues/106606

**Comment by rafiss** [comment]
@andyyang890 nice find! I think we can close this issue once we add a test to `master` that shows that this works as expected now.

**Comment by andyyang890** [comment]
Noting for the record that I ran a git bisect on the master branch for the small repro listed above and found that this was fixed by https://github.com/cockroachdb/cockroach/commit/99e29414fc2cc3986a7b6275adcdc9e3f0be412e, which is consistent with my observation that this is fixed starting in 22.2.
